DQM/Physics/src/ExoticaDQM.cc

```diff
@@ -572,11 +620,135 @@ void ExoticaDQM::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup
   analyzeMonoMuons(iEvent);
   analyzeMonoElectrons(iEvent);
 
-  //
-  //analyzeMultiJetsTrigger(iEvent);
-  //analyzeLongLivedTrigger(iEvent);
+  // Displaced Fermion Searches (requires MC truth)
+  if (ValidGenParticles) {
+    analyzeDisplacedLeptons(iEvent, iSetup);
+    analyzeDisplacedJets(iEvent, iSetup);
+  }
+
+}
 
+void ExoticaDQM::analyzeDisplacedLeptons(const Event & iEvent, const edm::EventSetup& iSetup){
+
+  //=== This is designed to run on MC events in which a pair of long-lived stop quarks each decay to a displaced lepton + displaced b jet.
+
+  const unsigned int stop1 = 1000006; // PDG identifier of top squark1
+  const unsigned int stop2 = 2000006; // PDG identifier of top squark2
+  const float deltaRcut = 0.01; // Cone size for matching reco to true leptons.
+  const float invPtcut = 0.1; // Cut in 1/Pt consistency for matching reco tracks to genParticles.
+ 
+  for (const reco::GenParticle& gen : *GenCollection_){
+    unsigned int idPdg = abs(gen.pdgId());
+    if (idPdg == stop1 || idPdg == stop2) {
+      unsigned int nDau = gen.numberOfDaughters();
+      for (unsigned int i = 0; i < nDau; i++) {
+        const reco::GenParticle* dau = (const reco::GenParticle*) gen.daughter(i);
```

Comment:
@davidlange6 - I see your point. These DQM plots were asked from the LongLived teams in EXO, who provided the code to be implemented within the existing ExoticaDQM.cc.
I was not aware of this being a problem. The real effect is that these plots will be only available for RelVal of SIM and not for DATA. If this is a big issue I could remove them for the time being. The rest of the PR is quite urgent as fixes some issues with DQM plots not shown in DATA.